src/components/edit/AreaCRUD.tsx

```diff
@@ -100,53 +59,24 @@ export const AreaCRUD = ({ uuid: parentUuid, areaName: parentName, childAreas, e
         </div>)}
 
       {areaCount > 0 && (
-        <DndContext
-          sensors={sensors}
-          collisionDetection={closestCenter}
-          onDragEnd={handleDragEnd}
-          onDragStart={handleDragStart}
-        >
-          <div
-            className={`two-column-table ${editMode ? '' : 'xl:grid-cols-2 lg:grid-cols-2 md:grid-cols-2'
-              }  fr-2`}
-          >
-            <SortableContext
-              items={areasSortedState}
-              strategy={rectSortingStrategy}
-            >
-              {areasSortedState.map((uuid, idx) => (
-                <SortableItem id={uuid} key={uuid} disabled={!editMode}>
-                  <AreaItem
-                    index={idx}
-                    borderBottom={[Math.ceil(areaCount / 2) - 1, areaCount - 1].includes(idx)}
-                    parentUuid={parentUuid}
-                    {/* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-                    ...areaStore.get(uuid)!}
-                    editMode={editMode}
-                    onChange={onChange}
-                  />
-                </SortableItem>
-              ))}
-            </SortableContext>
-          </div>
-          <DragOverlay>
-            {draggedArea != null
-              ? (
-                <div className='bg-purple-100'>
-                  <AreaItem
-                    index={areasSortedState.indexOf(draggedArea)}
-                    borderBottom
-                    parentUuid={parentUuid}
-                    {/* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-                    ...areaStore.get(draggedArea)!}
-                    editMode={editMode}
-                    onChange={onChange}
-                  />
-                </div>
-                )
-              : null}
-          </DragOverlay>
-        </DndContext>
+        <DraggableTable
+          rowIds={Array.from(areaStore.keys())}
+          editMode={editMode}
+          onDragEnd={onDragEnd}
+          renderRow={(areaId: string, idx: number) => {
```

Comment:
A little bit inelegant, but I don't really know of a better design pattern. We need to pass in both the child component and the data to fill it with. One other idea is to pass in a list of childProps.
```
<DraggableTable ..>
  <AreaItem><AreaItem ...>
<DraggableTable ..>
```
But then we would have to maintain extra state here in the parent component to order the child components, which is worse than slightly ugly code IMO.